internal/verify/enum.go

```diff
@@ -0,0 +1,41 @@
+package verify
+
+import (
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type EnumValues[T ~string] interface {
+	~string
+	Values() []T
```

Comment:
Interface could have been `Values() string` to provide a more idiomatic Go. This would allow a simple interface rather than generics.
This change require fixing sdk generation.